src/components/tag-filter/tag-filter.types.ts

```diff
@@ -0,0 +1,14 @@
+import { type Theme } from 'baseui';
+
+export type TagFilterOptionConfig = {
+  label: string;
+  startEnhancer?: React.ComponentType<{ theme: Theme }>;
+};
+
+export type Props<T extends string> = {
```

Comment:
What's the purpose of `T extends string`? 

Is it not sufficient to use `string` instead?

Reply:
I wanted to allow usage to specify as strict a type as necessary. In theory, a user could just create a tag filter with T as string, but they could also use something like "WorkflowStatus".

Reply:
Nice, thx!